modules/flyte/iam.tf

```diff
@@ -0,0 +1,102 @@
+resource "aws_iam_role" "flyte_controlplane" {
+  name = "${local.deploy_id}-flyte-controlplane"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRoleWithWebIdentity"
+        Effect = "Allow"
+        Principal = {
+          Federated = local.oidc_provider_arn
+        }
+        Condition : {
+          StringEquals : {
+            "${trimprefix(local.oidc_provider_url, "https://")}:aud" : "sts.amazonaws.com",
+            "${trimprefix(local.oidc_provider_url, "https://")}:sub" : [
+              "system:serviceaccount:domino-platform:flyteadmin",
```

Comment:
I wouldnt hardcode the namespaces name nor the sa names. This should be defined in the deployer and passed through to both the terraform and later on to the installer.

Reply:
Yeah I was planning on addressing that in a separate PR since it is tracked by a different [ticket](https://dominodatalab.atlassian.net/browse/DOM-54326), but I can incorporate it here if you'd prefer

Reply:
[Here](https://github.com/dominodatalab/terraform-aws-eks/pull/221) is the other PR, I can merge it into this one if that makes sense

Reply:
yes please.